io/src/test/scala/sbt/io/IOSpec.scala

```diff
@@ -70,6 +70,101 @@ class IOSpec extends FunSuite {
     assert(IO.getModifiedTimeOrZero(file("/not/existing/path")) == 0L)
   }
 
+  // classLocation -----------
+
+  test("classLocation[Integer] should return a URL pointing to the rt.jar or java.base") {
+    val u = IO.classLocation[java.lang.Integer]
+    assert(
+      (u.toString == "jrt:/java.base") ||
+        (u.toString.startsWith("file:") && u.toString.endsWith("rt.jar") && IO.asFile(u).isFile))
+  }
+
+  test(
+    "classLocation[AbstractMap.SimpleEntry] should return a URL pointing to the rt.jar or java.base") {
+    val u = IO.classLocation[java.util.AbstractMap.SimpleEntry[String, String]]
+    assert(
+      (u.toString == "jrt:/java.base") ||
+        (u.toString.startsWith("file:") && u.toString.endsWith("rt.jar") && IO.asFile(u).isFile))
+  }
+
+  test("classLocation[this.type] should return a URL pointing to a directory") {
+    val u = IO.classLocation[this.type]
+    assert(IO.asFile(u).isDirectory)
+  }
+
+  // classLocationPath -----------
+
+  test("classLocationPath[Integer] should return NIO path pointing to the rt.jar or java.base") {
```

Comment:
This is more of a comment on #181, but I think maybe there should be a classLocationPathOption. I'd assume that the IO.classLocation* methods are used more or less exclusively within the sbt project, but I could see a user "fixing" the deprecation warnings by replacing classLocationFile with classLocationPath and ultimately having runtime failures on jdk 11.

Reply:
I thought the NIO Path is able to represent module path, but not `File`.

Reply:
sbt/sbt would probably use `classLocationFileOption`.